kibana/templates/deployment.yaml

```diff
@@ -52,10 +52,39 @@ spec:
     {{- with .Values.nodeSelector }}
       nodeSelector:
 {{ toYaml . | indent 8 }}
-    {{- end }}
-    {{- with .Values.affinity }}
```

Comment:
The problem here is that the documentation is wrong, not the template itself. This behavior works and actually has template testing for it defined here: https://github.com/elastic/helm-charts/blob/007d6e4096c4419200b7c52c56781e9e1974c369/kibana/tests/kibana_test.py#L117-L133. Switching this over to match the Elasticsearch chart makes sense, however this would make it a breaking change for anybody that is correctly configuring this (by ignoring the wrong documentation). So I think the way forward here is to update the documentation not the template.